### PR TITLE
Improved logging and error checking in Experiments

### PR DESCRIPTION
**Pull request by skonefal**
Logs are more informative.

Experiments checks for errors when stopping Best Effort tasks.

### CODE REVIEW

**Comment by skonefal** [comment]
run integration tests

**Comment by skonefal** [comment]
run integration test

**Comment by skonefal** [comment]
run integration tests

File: experiments/memcached-cat/main.go [hunk 1]
```diff
@@ -75,11 +75,11 @@ func main() {
 
 	// Connect to metadata database
 	metadata, err := experiment.NewMetadata(uid, experiment.MetadataConfigFromFlags())
-	errutil.CheckWithContext(err, "Cannot connect to metadata database")
+	errutil.CheckWithContext(err, "Cannot connect to in Cassandra Metadata Database")
```
Comment (iwankgb) [inline experiments/memcached-cat/main.go]: @skonefal - get rid of "in", please.

File: experiments/memcached-cat/main.go [hunk 2]
```diff
@@ -278,7 +278,12 @@ func main() {
 							}
 						}
 
-						snapHandle, err := snapSession.LaunchSession(loadGeneratorHandle, snapTags)
+						err = beHandle.Stop()
+						if err != nil {
+							return errors.Wrapf(err, "best effort task has failed in phase %s", phaseName)
```
Comment (iwankgb) [inline experiments/memcached-cat/main.go]: @skonefal - has task failed or task stopping?

File: experiments/memcached-sensitivity-profile/main.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ func main() {
 
 	metadata, err := experiment.NewMetadata(uid, experiment.MetadataConfigFromFlags())
 	if err != nil {
-		logrus.Errorf("Cannot connect to metadata database %q", err.Error())
+		logrus.Errorf("Cannot connect to in Cassandra Metadata Database %q", err.Error())
```
Comment (iwankgb) [inline experiments/memcached-sensitivity-profile/main.go]: @skonefal - get rid of "in".

File: experiments/optimal-core-allocation/main.go [hunk 4]
```diff
@@ -62,11 +62,11 @@ func main() {
 
 	// connect to metadata database
 	metadata, err := experiment.NewMetadata(uid, experiment.MetadataConfigFromFlags())
-	errutil.CheckWithContext(err, "Cannot connect to metadata database")
+	errutil.CheckWithContext(err, "Cannot connect to in Cassandra Metadata Database")
```
Comment (iwankgb) [inline experiments/optimal-core-allocation/main.go]: @skonefal - get rid of "in".

File: experiments/memcached-sensitivity-profile/main.go [hunk 5]
```diff
@@ -230,6 +231,11 @@ func main() {
 						}
 					}
 
+					err = beHandle.Stop()
+					if err != nil {
+						return errors.Wrapf(err, "best effort task has failed in phase %s, repetition %d", phaseName, repetition)
```
Comment (iwankgb) [inline experiments/memcached-sensitivity-profile/main.go]: @skonefal - has task failed or task stopping?

File: experiments/specjbb-sensitivity-profile/main.go [hunk 6]
```diff
@@ -207,6 +208,11 @@ func main() {
 					}
 					loadGeneratorHandle.Wait(0)
 
+					err = beHandle.Stop()
+					if err != nil {
+						return errors.Wrapf(err, "best effort task has failed in phase %s", phaseName)
```
Comment (iwankgb) [inline experiments/specjbb-sensitivity-profile/main.go]: @skonefal - has task failed or task stopping?

File: experiments/memcached-cat/main.go (hunk 2 above)
Comment (skonefal) [inline experiments/memcached-cat/main.go]: Failed, as fail in ServiceHandle.

File: experiments/memcached-sensitivity-profile/main.go (hunk 5 above)
Comment (skonefal) [inline experiments/memcached-sensitivity-profile/main.go]: Failed, as fail in ServiceHandle.

File: experiments/specjbb-sensitivity-profile/main.go (hunk 6 above)
Comment (skonefal) [inline experiments/specjbb-sensitivity-profile/main.go]: Failed, as fail in ServiceHandle.

**Comment by skonefal** [comment]
run integration tests

**Comment by skonefal** [comment]
run integration tests

**Pull request merged**